deferral/ah-django
# 164859745 Document the api endpoints with swagger

### What does this PR do?

* Documents the API endpoints using swagger

### Description of Task to be completed?

- [x] Install the drf-yasg dependency to generate swagger schemas
- [x] Define url rules for the swagger documentation link

### How should this be manually tested?

1. Clone the repo to your local machine from [here](https://github.com/deferral/ah-django.git)
2. cd into the ah-django directory and checkout to the branch ```ft-swagger-docs-164859745```

       cd ah-django

       git checkout ft-swagger-docs-164859745

2. Create a virtual environment and activate it:

       python3 -m virtualenv venv && source venv/bin/activate

 * If you need to install virtualenv first:

       pip install virtualenv

4. Install the project dependencies

       pip install -r requirements.txt

5. Make database migrations and start the django server

       python3 manage.py makemigrations

       python3 manage.py migrate
 
       python3 manage.py runserver 

6. In your browser, test the url ```http://127.0.0.1:8000/apidocs```

You should be able to see a swagger ui with the endpoint in the application

### Any background context you want to provide?

Initially, the application was not documented

### What are the relevant pivotal tracker stories?

[#164859745](https://www.pivotaltracker.com/story/show/164859745)

![Screenshot from 2019-03-27 18-35-26](https://user-images.githubusercontent.com/38868860/55089843-23234380-50bf-11e9-8595-aec772aa6256.png)
